ot3_state_manager/ot3_state_manager/util.py

```diff
@@ -0,0 +1,108 @@
+"""Functions and classes that are shared across the package."""
+
+from __future__ import annotations
+
+import enum
+import hashlib
+from dataclasses import dataclass
+from typing import Literal, Optional, Union
+
+from opentrons.hardware_control.types import OT3Axis
+
+
+class Direction(enum.Enum):
+    """Class representing direction to move axis in."""
+
+    POSITIVE = "+"
+    NEGATIVE = "-"
+
+    @classmethod
+    def from_symbol(cls, symbol: str) -> "Direction":
+        """Parse plus or minus symbol into direction."""
+        if symbol == "+":
+            return cls.POSITIVE
+        elif symbol == "-":
+            return cls.NEGATIVE
+        else:
+            raise ValueError(
+                f'Symbol "{symbol}" is not valid.' f' Valid symbols are "+" and "-"'
+            )
+
+
+@dataclass
+class Message:
+    """Generic Message Type."""
+
+    @classmethod
+    def to_message(cls, string: str) -> Optional[Message]:
+        """Parse string into message."""
+        raise NotImplementedError
+
+
+@dataclass
+class SyncPinMessage(Message):
+    """Message for setting sync pin high or low."""
+
+    state: Literal["HIGH", "LOW"]
+
+    @classmethod
+    def to_message(cls, string: str) -> Optional[SyncPinMessage]:
+        """Parse string into SyncPinMessage or None."""
+        if string.startswith("SYNC_PIN"):
+            split_string = string.split(" ")
+            if len(split_string) == 2:
+                sync_pin_state = split_string[1]
+                if sync_pin_state == "HIGH":
+                    return cls(state="HIGH")
+                elif sync_pin_state == "LOW":
+                    return cls(state="LOW")
+
+        return None
+
+
+@dataclass
+class MoveMessage(Message):
```

Comment:
technically the sync pin would also have its state change as a result of a move -- should we incorporate that into a move message?

Reply:
I can do that. How does it work?